src/Data/Functor/Contravariant/Compose.purs

```diff
@@ -0,0 +1,36 @@
+module Data.Functor.Contravariant.Compose where
+
+import Data.Functor.Contravariant
+
+-- | Composition of two contravariant functors.
+newtype Compose f g a = Compose (f (g a))
+
+runCompose :: forall f g a. Compose f g a -> f (g a)
+runCompose (Compose f) = f
+
+instance functorCompose :: (Contravariant f, Contravariant g) => Functor (Compose f g) where
+  (<$>) f (Compose x) = Compose ((f >$<) >$< x)
+
+-- | Composition of a covariant and contravariant functor.
+newtype ComposeFC f g a = ComposeFC (f (g a))
```

Comment:
Good point, having both versions might be unnecessary. Will give it a try after moving this out.